packages/utils/src/lib/transform.ts

```diff
@@ -156,3 +157,56 @@ export function toOrdinal(value: number): string {
   return `${value}th`;
 }
 /* eslint-enable no-magic-numbers */
+
+export class AuditsNotImplementedError extends Error {
+  constructor(list: { slug: string }[], auditSlugs: string[]) {
+    super(
+      `audits: "${auditSlugs
+        .filter(slug => !list.some(a => a.slug === slug))
+        .join(', ')}" not implemented`,
+    );
+  }
+}
+
+export function filterByAuditSlug<
+  T extends {
+    refs: GroupRef[];
+  },
+  L extends T[],
+  S extends L[number]['refs'][number]['slug'],
+>(groups: L, auditSlugs: S | S[]): L {
+  const slugs = toArray(auditSlugs);
+  if (slugs.length === 0) {
+    return groups;
+  }
+  return (
+    groups
+      // filter out groups that have no audits includes from onlyAudits (avoid empty groups)
+      .filter(group => group.refs.some(({ slug }) => slugs.includes(slug as S)))
+      .map(group => {
+        const groupsRefs = group.refs.filter(({ slug }) =>
+          slugs.includes(slug as S),
+        );
+
+        return {
+          ...group,
+          refs: groupsRefs,
+        };
+      }) as L
+  );
+}
+
+export function filterBySlug<T extends Audit, L extends T[]>(
+  list: L,
+  auditSlugs: L[number]['slug'][] | L[number]['slug'],
+): L {
+  const slugs = toArray(auditSlugs);
+  if (slugs.length === 0) {
+    return list;
+  }
+  if (slugs.some(slug => !list.some(wS => wS.slug === slug))) {
+    throw new AuditsNotImplementedError(list, slugs);
+  }
+
+  return list.filter(({ slug }) => slugs.includes(slug)) as L;
+}
```

Comment:
As mentioned above, the audit slug isn't union but a plain old `string`. So there's no advantage to the generic type parameters here, it just adds complexity:

```suggestion
export function filterByAuditSlug(
  groups: Group[],
  auditSlugs: string | string[],
): Group[] {
  const slugs = toArray(auditSlugs);
  if (slugs.length === 0) {
    return groups;
  }
  return (
    groups
      // filter out groups that have no audits includes from onlyAudits (avoid empty groups)
      .filter(group => group.refs.some(({ slug }) => slugs.includes(slug)))
      .map(group => {
        const groupsRefs = group.refs.filter(({ slug }) =>
          slugs.includes(slug),
        );

        return {
          ...group,
          refs: groupsRefs,
        };
      })
  );
}

export function filterBySlug(
  list: Audit[],
  auditSlugs: string | string[],
): Audit[] {
  const slugs = toArray(auditSlugs);
  if (slugs.length === 0) {
    return list;
  }
  if (slugs.some(slug => !list.some(wS => wS.slug === slug))) {
    throw new AuditsNotImplementedError(list, slugs);
  }

  return list.filter(({ slug }) => slugs.includes(slug));
}
```